### PR TITLE
Add OrderQuery and MemberQuery to GraphQL v2

### PR DESCRIPTION
**Pull request by sbinlondon**
* Adds `OrderQuery` and `MemberQuery` for v2
* Adds `fetchOrderWithReference`

### CODE REVIEW

**Comment by znarf** [comment]
@sbinlondon What's the use case for MemberQuery? I'm not sure about it. `fromAccount` and `toAccount` are looking odd.

**Comment by sbinlondon** [comment]
https://github.com/opencollective/opencollective-frontend/blob/471b43ab7e5fb9bdbd3f97b7763c06c4a85ba643/components/new-contribution-flow/ContributionFlowSuccess.js#L337

When we navigate to the order success page, we don't feed in any order information via props. We do have the `OrderId` in the router query though. So first we get the order info using the `OrderQuery` but then we also need the `Member` info to see if there is a public message from when the sponsor has donated to this collective before. So we make an additional `MemberQuery` using the info we get back from the `OrderQuery` (the `toAccount`, or the Collective that's being donated to, and the `fromAccount`, who is giving the donation).

There is a v1 query I could use from the old Order page but I thought we were going to update old queries as we went.

**Comment by znarf** [comment]
I think that could be introduced as a property to the `Order` like `order.membership`. Would be interested to see @Betree's feedback on this.

**Comment by Betree** [comment]
> I think that could be introduced as a property to the Order like order.membership. Would be interested to see @Betree's feedback on this.

I agree with this. We could also add a `account` field on `account.memberOf`:

```gql
order(order: $order) {
  fromAccount {
    memberOf(account: $account) {
      // Returns all the memberships for fromAccount <> account
    }
  }
}
```

Whatever we do, it would be elegant to solve this with just one query.

**Comment by sbinlondon** [comment]
@Betree @znarf I've made some changes, let me know what you think

File: server/graphql/v2/interface/IsMemberOf.js [hunk 1]
```diff
@@ -59,6 +61,10 @@ export const IsMemberOfFields = {
           [Op.in]: args.accountType.map(value => AccountTypeToModelMapping[value]),
         };
       }
+      if (args.account) {
+        const account = await fetchAccountWithReference(args.account);
```
Comment (Betree) [inline server/graphql/v2/interface/IsMemberOf.js:65]: Small perf improvement

```suggestion
        const account = await fetchAccountWithReference(args.account, { loaders: req.loaders });
```

**Review by Betree — APPROVED** [review]
Works for me!

**Pull request merged**